Feature/set models properties

zhihdd opened the pull request:
每個model關聯的設定已設定好，
local 端test也有通過(為通過test，關聯設定與教案有些不同)
model的關聯建立請多幫忙留意。


ZHIH

zhihdd committed:
feature: add attributes of followships

zhihdd committed:
feature: add attributes of Like

zhihdd committed:
feature: add attributes of Reply

zhihdd committed:
feature: add attributes of Tweet

zhihdd committed:
feature: add attributes of User

zhihdd committed:
fix: remove notNull from attribute of deletedAt、UserId、tweetId、lastLogin and fix models test file back

zhihdd committed:
feature: add User association

1.(1-M)with Like

2.(1-M)with Reply

3.(1-M)with Tweet

(M-M)with Followship

zhihdd committed:
feature: add Tweet association

1.(1-M)with Like

2.(1-M)with Reply

3.(M-1)with User

zhihdd committed:
feature: add Reply association

1.(M-1)with Tweet

2.(M-1)with User

zhihdd committed:
feature: add Like association

1.(M-1)with Tweet

2.(M-1)with User

zhihdd committed:
feature: Add model of Admin and association with Tweet

Jennesy commented on migrations/20190115071420-create-reply.js:
```diff
@@ -1,30 +1,33 @@
         autoIncrement: true,
         primaryKey: true,
-        type: Sequelize.INTEGER
+        type: Sequelize.INTEGER,
```
是不是不小心加到逗號

zhihdd replied:
這我預設格式會幫我處理，我再刪就好

Jennesy commented on migrations/20190115071420-create-reply.js:
```diff
@@ -1,30 +1,33 @@
-        type: Sequelize.INTEGER
+        type: Sequelize.INTEGER,
       },
       comment: {
```
我們ERD和API document上attribute是命名為content，可以討論一下要命名為"comment"還是"content"。

zhihdd replied:
OK我照內建的migration 的create-Reply去做調整的 ，但剛看測試檔應該不影響結果，所以我改成"content"好了，這樣文件
也不用動。

Jennesy commented on migrations/20190115071420-create-tweet.js:
```diff
@@ -1,27 +1,39 @@
-        type: Sequelize.INTEGER
+        type: Sequelize.INTEGER,
       },
       description: {
```
命名跟ERD與API document不一致～

zhihdd replied:
同上可改~~

Jennesy commented on models/tweet.js:
```diff
@@ -1,8 +1,48 @@
+  );
+  Tweet.associate = function (models) {
+    Tweet.hasMany(models.Reply);
+    Tweet.hasMany(models.Like);
```
可使用
`Tweet.belongsToMany(models.User, {
      through: models.Like,
      foreignKey: "id",
      as: "LikedUsers",
    })`
    建立tweet與user間的多對多關係，如followship

zhihdd replied:
這邊會這樣用的原因是，測試檔逼我的 QAQ (ry，詳見 test/models/Tweet.pec.js
```
    it('should have many replies', (done) => {
      expect(Tweet.hasMany).to.have.been.calledWith(Reply)
      done()
    })
    it('should have many likes', (done) => {
      expect(Tweet.hasMany).to.have.been.calledWith(Like)
      done()
    })
    it('should belong to user', (done) => {
      expect(Tweet.belongsTo).to.have.been.calledWith(User)
      done()
    })
 
```
如果不那樣測試不給過

Jennesy replied:
🥲
我想到一個方法，就是兩個都放。
`Tweet.hasMany(models.Like)`
`Tweet.belongsToMany(models.User, { through: models.Like, foreignKey: "id", as: "LikedUsers", }) `
同時都放⬆️

zhihdd replied:
不知道會不會衝突，不過可以放放看www，我等等放

Jennesy commented on models/user.js:
```diff
@@ -1,8 +1,75 @@
+    {}
+  );
+  User.associate = function (models) {
+    User.hasMany(models.Like);
```
同理，可使用
`User.belongsToMany(models.Tweet, { through: models.Like, foreignKey: "id", as: "LikedTweets", })`
來建立多對多關係，如followship

zhihdd replied:
同上(ry QWQ

Jennesy reviewed (changes requested):
統一整理：
1. 更改Reply與Tweet model ‘content’參數的命名
2. 嘗試加User和Tweet `belongsToMany` Like的association

zhihdd committed:
fix: modify attributes and associtation

1.descripotion change to content in model Tweet

2.comment change to content in model Reply

3.add belongstoMany in model User and Tweet

Jennesy reviewed (approved):
Good!
